src/common.c

```diff
@@ -78,14 +78,30 @@ int fi_poll_fd(int fd, int timeout)
 	return ret == SOCKET_ERROR ? -ofi_sockerr() : ret;
 }
 
+/*
+ * Find the location of the most significant bit set. Value 1~64
+ * means such bit is found. Value 0 means no bit is set.
+ */
+static inline int msb(uint64_t x)
+{
+	int pos = 0;
+
+	while (x) {
+		x >>= 1;
+		pos++;
+	}
+
+	return pos;
+}
+
 uint64_t fi_tag_bits(uint64_t mem_tag_format)
 {
-	return UINT64_MAX >> (ffsll(htonll(mem_tag_format)) -1);
+	return UINT64_MAX >> (64 - msb(mem_tag_format));
 }
 
 uint64_t fi_tag_format(uint64_t tag_bits)
```

Comment:
And this function is treating the input as a max_tag_value, rather than using it as some number of bits.  Maybe we can rename the parameter?

Reply:
Yes we can rename the parameter as max_tag_value. In the psm2 provider, the parameter passed to the call is actually named that way.

While we are here, it may also make sense to change the function name to ofi_tag_format() for consistency.

Reply:
Thanks - that makes sense too.